### ISSUE TITLE
"This is not yet production ready"

### ISSUE THREAD

**Issue by jonathanstrong** [Nemo157/roaring-rs#67]
hello,

Thank you for publishing this library and all the hard work you've put into it!

I was wondering if you could possibly offer additional explanation regarding the statement in the documentation, "This is not yet production ready." Specifically, what aspects of the library do you consider not to be production ready?

thank you!

Jonathan

**Comment by lemire** [comment]
For context, this remark in the README was written 6 years ago.

**Comment by Nemo157** [comment]
Yeah, this was the first Rust code I ever wrote (started just before Rust 1.0 came out :grin:) so I wanted to temper any expectations on it. I think in terms of _functionality_ it is fine to use, what is there should work (well, hopefully, #62). The biggest things I have not got around to doing since I have become more familiar with Rust are:

 1. A complete API review. When this was started the API norms weren't as strong as they are now, so there are likely to be small inconsistencies in the API such as #45, and new APIs are available to implement such as #5.
 2. Benchmarking and optimization. A long time ago I started a benchmark suite comparing against `croaring`: https://github.com/Nemo157/roaring-bench-rs but I never got around to actually using it to optimize anything.
